Release_Process.md

```diff
@@ -42,7 +43,8 @@ These steps are common between minor and patch releases:
    - Target: the release branch that was just pushed
    - Title: Same as tag name
    - Description: The body of the changelog entry (Added, Changed etc)
-1. Publish the release on Github
+1. Publish the release on GitHub
+1. (Optional) Do a `make clean` in the repository root directory (`bigchaindb/`)
```

Comment:
Good points and ideas:  I'll add a line about checking out the tag generated by the GitHub release, and a line about `make release`.